comp/dogstatsd/listeners/udp_test.go

```diff
@@ -224,14 +224,13 @@ func TestUDPReceive(t *testing.T) {
 	telemetryStore := NewTelemetryStore(nil, deps.Telemetry)
 	packetsTelemetryStore := packets.NewTelemetryStore(nil, deps.Telemetry)
 	s, err := NewUDPListener(packetChannel, newPacketPoolManagerUDP(deps.Config, packetsTelemetryStore), deps.Config, nil, telemetryStore, packetsTelemetryStore)
-	require.NotNil(t, s)
 	assert.Nil(t, err)
```

Comment:
Would it make sense to change this to require? This will make the test fail here, rather than some time later and make it easier to find the relevant error message in the logs.

Reply:
I have the test fails a step further so we can gather a bit more loggings from NewUDPListener, currently if s is not nil, we do not get the bubbled up errors from the function [here](https://github.com/DataDog/datadog-agent-macos-build/actions/runs/9386899678/job/25848556728).

<img width="1113" alt="image" src="https://github.com/user-attachments/assets/6ba487da-36df-4915-bdb9-065f7962fd18">

Reply:
What I meant was that instead of assert.Nil on error, we do this:

```suggestion
	require.NoError(t, err)
```

So the test stops with the error, and not with the nil assertion on the listener.